### PR TITLE
Searchsource setting

### PR DESCRIPTION
**Pull request by matrottier**
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines:https://github.com/infra-geo-ouverte/igo2/blob/master/.github/CONTRIBUTING.md#git-commit-guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)
#349 

**What is the new behavior?**
control which source-type can be queried

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x] No
```

If this PR contains a breaking change, please describe the impact and migration path for existing applications:

**Other information**:

### CODE REVIEW

File: packages/geo/src/lib/search/search-selector/search-selector.component.html [hunk 1]
```diff
@@ -24,5 +23,65 @@
...
+      mat-menu-item
+      [matMenuTriggerFor]="setting">
+      <mat-icon svgIcon="settings"></mat-icon>{{ 'igo.geo.search.settings' | translate}}
+    </button>
   </mat-menu>
+
+  <mat-menu
+    #setting="matMenu"
+    class="no-border-radius">
+      <ng-container *ngFor="let source of getSearchSources()">
```
Comment (cbourget) [inline packages/geo/src/lib/search/search-selector/search-selector.component.html]: Could this be a component on his own?

File: packages/geo/src/lib/search/search-selector/search-selector.component.ts [hunk 2]
```diff
@@ -1,3 +1,5 @@
+import {MatDialog} from '@angular/material';
```
Comment (cbourget) [inline packages/geo/src/lib/search/search-selector/search-selector.component.ts]: Is this used?

File: packages/geo/src/lib/search/search-selector/search-selector.component.ts [hunk 3]
```diff
@@ -82,4 +86,44 @@ export class SearchSelectorComponent implements OnInit {
...
+   */
+  getSearchSources(): SearchSource[] {
+    return this.searchSourceService.getSources();
+  }
+
+  /**
+   * Triggered when a setting is checked (checkbox style)
+   * @internal
+   */
+  settingsValueCheckedCheckbox(event, source, setting, settingValue) {
```
Comment (cbourget) [inline packages/geo/src/lib/search/search-selector/search-selector.component.ts]: Type is missing

File: packages/geo/src/lib/search/search-selector/search-selector.component.ts [hunk 4]
```diff
@@ -82,4 +86,44 @@ export class SearchSelectorComponent implements OnInit {
...
+    settingValue.enabled = event.checked;
+    source.setParamFromSetting(setting);
+  }
+
+  /**
+   * Triggered when a setting is checked (radiobutton style)
+   * @internal
+   */
+  settingsValueCheckedRadioButton(event, source, setting, settingValue) {
```
Comment (cbourget) [inline packages/geo/src/lib/search/search-selector/search-selector.component.ts]: Same

File: packages/geo/src/lib/search/search-selector/search-selector.component.html (hunk 1 above)
Comment (matrottier) [reply packages/geo/src/lib/search/search-selector/search-selector.component.html]: Yes, in the future, a more elaborated menu in a "dialog" could be a component.   
That was my first track. 
This was a first draft and get something fast and easy to allow the user to choose some service parameter.

File: packages/geo/src/lib/search/search-selector/search-selector.component.ts (hunk 2 above)
Comment (matrottier) [reply packages/geo/src/lib/search/search-selector/search-selector.component.ts]: That was my first track
wreckage ...

File: packages/geo/src/lib/search/search-selector/search-selector.component.html (hunk 1 above)
Comment (cbourget) [reply packages/geo/src/lib/search/search-selector/search-selector.component.html]: I would love to see it in it's own component (a dialog or the same thing) but it's not my call..

Comment (matrottier) [reply packages/geo/src/lib/search/search-selector/search-selector.component.html]: that make sense, a "search-settings" component for example.  
"search-selector" gonna be inside "search-settings" ?  
or configurable presence on the "search-bar" ?  
@mbarbeau welcome on board !

File: packages/geo/src/lib/search/search-selector/search-selector.component.ts [hunk 5]
```diff
@@ -1,3 +1,5 @@
+import {MatCheckboxChange, MatRadioChange } from '@angular/material';
```
Comment (cbourget) [inline packages/geo/src/lib/search/search-selector/search-selector.component.ts]: Are these import needed now that the settings have their own component?

File: packages/geo/src/lib/search/search-selector/search-selector.component.ts [hunk 6]
```diff
@@ -9,6 +11,8 @@ import {
 
 import { SEARCH_TYPES } from '../shared/search.enums';
 import { SearchSourceService } from '../shared/search-source.service';
+import { SearchSource } from '../shared/sources/source';
+import { SearchSourceSettings, SettingOptions } from '../shared/sources/source.interfaces';
```
Comment (cbourget) [inline packages/geo/src/lib/search/search-selector/search-selector.component.ts]: Same

File: packages/geo/src/lib/search/search-selector/search-selector.module.ts [hunk 7]
```diff
@@ -6,7 +6,9 @@ import {
   MatIconModule,
   MatButtonModule,
   MatMenuModule,
-  MatRadioModule
+  MatRadioModule,
```
Comment (cbourget) [inline packages/geo/src/lib/search/search-selector/search-selector.module.ts]: Same

File: packages/geo/src/lib/search/search-settings/search-settings.component.ts [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+export class SearchSettingsComponent implements OnInit {
+
+  /**
+   * Event emitted when the enabled search type changes
+   */
+  @Output() change = new EventEmitter<string>();
+
+  constructor(private searchSourceService: SearchSourceService) {}
+
+  /**
```
Comment (cbourget) [inline packages/geo/src/lib/search/search-settings/search-settings.component.ts]: Not needed

File: packages/geo/src/lib/search/shared/search-source.service.ts [hunk 9]
```diff
@@ -39,4 +39,8 @@ export class SearchSourceService {
       }
     });
   }
+
+  setParamFromSetting(source, setting) {
```
Comment (cbourget) [inline packages/geo/src/lib/search/shared/search-source.service.ts]: Type missing

File: packages/geo/src/lib/search/shared/sources/source.ts [hunk 10]
```diff
@@ -83,6 +84,37 @@ export class SearchSource {
...
+        case 'radiobutton':
+          setting.values.forEach( conf => {
+            if (conf.enabled) {
+              this.options.params = Object.assign( (this.options.params || {}),
+                                                    { [setting.name] : conf.value } );
+            }
+          });
+          break;
+        case 'checkbox':
+          let confValue = '';
```
Comment (cbourget) [inline packages/geo/src/lib/search/shared/sources/source.ts]: You could do something like this. You wouldn't have to remove the tailing comma.

```
const values = [];
values.push(conf.value);
const value = values.join(',');
```

or
```
const value = setting.values
  .filter((conf) => conf.enabled)
  .map((conf) => conf.value)
  .join(',')
```

**Pull request merged**